libs/x-content/src/main/java/org/elasticsearch/xcontent/Text.java

```diff
@@ -36,31 +31,45 @@ public static Text[] convertFromStringArray(String[] strings) {
         return texts;
     }
 
-    private BytesReference bytes;
+    private EncodedBytes bytes;
     private String text;
```

Comment:
(I wonder why this is called `text` when we refer to it as "string" everywhere else?)